### DISCUSSION TITLE
Infinite loop to previous segment timestamp

### DISCUSSION THREAD

**Discussion by dodysw3** [openai/whisper#810] [Q&A] [answered]
Hi, sometimes whisper transcription stuck in an infinite loop where it went to past segment timestamp and never get out of it. E.g. 
```
...
[38:40.140 --> 38:49.740]  Terus ada beberapa fitur juga di dalamnya yang sangat-sangat support sekali untuk di tahun 2022 ini.
[38:49.740 --> 38:34.860]  Di mana, Desk
[38:34.860 --> 38:40.140]  Tapi untuk resolusi 2K pun itu udah bagus, cuma ada yang lebih bagus lagi itu 4K.
[38:40.140 --> 38:49.740]  Terus ada beberapa fitur juga di dalamnya yang sangat-sangat support sekali untuk di tahun 2022 ini.
[38:49.740 --> 38:34.860]  Di mana, Desk
...
```
whisper parameters:
`whisper input.wav --language id --model medium --condition_on_previous_text False --no_speech_threshold 0.4 --best_of None --beam_size None --device cuda:0`

Any idea?

**Comment by junkoran** [comment]
solved blm bang?

**Comment by andrewchernyh** [comment, marked as answer]
I've catched something simular, and as I see it's related to https://github.com/openai/whisper/blob/5c1a8c10e762bf9c29fcf6b3e40f17bc8ab09864/whisper/transcribe.py#L216

When decoder has misprediction of the latest consecutive timestamp, in my case it was `tokenizer.timestamp_begin` seek changes is wrong and transcribe can't be finished.

I'm not sure, but I think decoder should suppress emitting of timestamp tokens with ts <= latest timestamp.
Something like https://github.com/openai/whisper/blob/5c1a8c10e762bf9c29fcf6b3e40f17bc8ab09864/whisper/decoding.py#L430, but according to the current state

**Reply by andrewchernyh** [reply]
Fix in https://github.com/openai/whisper/pull/914

**Comment by doublex** [comment]
Still fails: https://github.com/openai/whisper/discussions/924

**Comment by Jeronymous** [comment]
This problem is still here even after the fix by @andrewchernyh 
I report precisely with @doublex in https://github.com/openai/whisper/discussions/924
